File: easy/valid-mountain-array/valid-mountain-array.js
/**
 * @param {number[]} A
 * @return {boolean}
 */
var validMountainArray = function (A) {
  if (A.length < 3) {
    return false;
  }
  let hi = A[0];
  let hiIndex = 0;
  let increasing = true;
  for (let i = 1; i < A.length; i++) {
    if (increasing) {
      if (A[i] > hi) {
        hi = A[i];
        hiIndex = i;
      } else if (A[i] < hi) {
        increasing = false;
      } else {
        return false;
      }
    } else {
      if (A[i] >= A[i - 1]) {
        return false;
      }
    }
  }
  if (hiIndex === 0 || hiIndex === A.length - 1) {
    return false;
  }
  return true;
};

File: easy/valid-mountain-array/valid-mountain-array.c


bool validMountainArray(int *A, int ASize)
{
    int i = 0;

    while (i + 1 < ASize && A[i] < A[i + 1])
        i++;

    if (i == 0 || i == ASize - 1)
        return false;

    while (i + 1 < ASize && A[i] > A[i + 1])
        i++;

    return i == ASize - 1;
}